src/main/java/net/neoforged/neoforge/client/model/AbstractUnbakedModel.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.model;
+
+import net.minecraft.client.renderer.block.model.ItemTransforms;
+import net.minecraft.client.renderer.block.model.TextureSlots;
+import net.minecraft.client.resources.model.ModelBaker;
+import net.minecraft.client.resources.model.ModelState;
+import net.minecraft.client.resources.model.UnbakedModel;
+import net.minecraft.util.context.ContextMap;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Base unbaked model for custom models which support the standard top-level model parameters
+ * added by vanilla and NeoForge except elements but create the quads from something other
+ * than the vanilla elements spec.
+ */
+public abstract class AbstractUnbakedModel implements ExtendedUnbakedModel {
+    /**
+     * Holds the standard top-level model parameters except elements.
+     * {@link UnbakedModel#bake(TextureSlots, ModelBaker, ModelState, boolean, boolean, ItemTransforms, ContextMap)}
+     * must always use the values given as parameters instead of accessing this parameter directly.
```

Comment:
Should mention why (i.e. parent or child models might affect the properties, and that is only reflected in the parameters of `bake`).